### ISSUE TITLE
Revamp OCFWeb

### ISSUE THREAD

**Issue by nikhiljha** [ocf/ocfweb#597]
Paging @BernardZhao on this one... we need your rant.

Apologies if this is already documented somewhere, but I couldn't find it.

- https://www.ocf.berkeley.edu/about/staff looks super nice and should probably be the homepage
- https://www.ocf.berkeley.edu/docs/services/lab/ doesn't have a picture of the lab? but the staff page does? ??? ? ?? ? ?
- Why is https://www.ocf.berkeley.edu/staff-hours not a calendar?
- Why you want to join the OCF isn't only "below the fold" - it's literally at the bottom of the page.

<img width="1242" alt="Screen Shot 2020-05-07 at 6 42 04 PM" src="https://user-images.githubusercontent.com/2773700/81361147-73c2e980-9092-11ea-87e5-d3f075572ee3.png">

**Comment by dkess** [comment]
I can speak to some of these as well:

1. Staff recruitment page shouldn't be the homepage because our target audience is users of our services. There could be more room to make recruitment more prominent on the homepage, but don't forget about actual users.
2. Sure, this is an easy PR :)
3. I've seen efforts to turn the staff hours page into a calendar but it's a tricky UI problem. Staff hours should really be held weekly on a somewhat regular schedule, so imo it should stay as a day-of-week thing (regardless of how it's displayed)
4. Sure, this page could be re-organized, make the PR

There have been a lot of efforts to "revamp" ocfweb and they have all failed. Instead of envisioning big, sweeping changes I would recommend finding the smallest units of improvements you can and doing those incrementally. You can also try pulling things out of ocfweb and into their own projects (e.g. printlist). If you have big ideas that can't be split up, write up a design doc and get buy-in. That way you don't have to waste effort and you can even get other people to help you.

**Comment by BernardZhao** [comment]
@nikhiljha dkess gave some great points. I just wanted to comment on #3, which I believe is the hardest one of these things to do, since there's no way we want to roll our own calendar view, and I haven't found an easy component to do so.

Personally, I think that pulling things out of ocfweb like dkess said is a great choice, and probably how we should progress incrementally. For example, I think one of the things I would like to break off of ocfweb is its api. It's small, only really is depended on by one or two things atm, and could get better and gain room to grow when placed into its own project.

**Comment by cg505** [comment]
The website is tricky in general because the VAST majority of people that visit it are people that want to use the lab. After that is people wanting to use other OCF services. After THAT is potential and current staff. So things that might seem like they make a lot of sense actually don't. We just need to keep in mind the target audience. There are a LOT of things that could be improved with the site.

**Comment by nikhiljha** [comment]
Makes sense. There are still definitely some usability issues from a "people wanting to use the lab/host" perspective, so I'll keep this open for discussion & I'll try to think of a way to include 2 and 4.
